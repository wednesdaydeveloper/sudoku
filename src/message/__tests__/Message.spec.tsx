import * as React from 'react';
import MessageComponent, {Props} from '../Message';
import {shallow} from 'enzyme';

describe('Message', () => {

  let props: Props;

  beforeEach(() => {
    //  デフォルトのプロパティ
    props = {
        messages: [],
    };
  });

  it('rendering messagesが空', () => {
    const wrapper = shallow(<MessageComponent {...props} />);
    expect(wrapper.find('div').length).toBe(1);
    expect(wrapper.find('div').at(0).prop('className')).toBe('alert');
  });

  it('rendering messages２件', () => {
    props.messages = [
      {messageType: 'alert-success', messageString: '成功' },      
      {messageType: 'alert-fail', messageString: '失敗' },      
    ];

    const wrapper = shallow(<MessageComponent {...props} />);
    expect(wrapper.find('div').length).toBe(3);
    expect(wrapper.find('div').at(0).prop('className')).toBe('alert');
    
    expect(wrapper.find('div').at(1).prop('className')).toBe('alert alert-success');
    expect(wrapper.find('div').at(1).prop('children')).toBe('成功');
    expect(wrapper.find('div').at(2).prop('className')).toBe('alert alert-fail');
    expect(wrapper.find('div').at(2).prop('children')).toBe('失敗');
  });

});