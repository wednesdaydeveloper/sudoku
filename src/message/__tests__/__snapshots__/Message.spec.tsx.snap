// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MessageComponent の Snapshot +++capturing Snapshot of Home 1`] = `
<div
  className="alert"
>
  <div
    className="alert alert-success"
    role="alert"
  >
    成功
  </div>
  <div
    className="alert alert-fail"
    role="alert"
  >
    失敗
  </div>
</div>
`;
